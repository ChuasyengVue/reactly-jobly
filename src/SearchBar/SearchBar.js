import React, { useState } from 'react';

const SearchBar = ({ onSearch }) => {
    
    const [search, setSearch] = useState("");

    const handleChange = (evt) => {
        setSearch(evt.target.value);
    }

    const handleSubmit = (evt) => {
        evt.preventDefault();
        onSearch(search);
    }

    return(
        <div>

            <form onSubmit={handleSubmit}>
                <input
                type='text'
                name='search'
                placeholder='Enter search term..'
                value={search}
                onChange={handleChange}/>

                <button type='submit'>Submit</button>
            </form>
        </div>
    )
}

export default SearchBar;