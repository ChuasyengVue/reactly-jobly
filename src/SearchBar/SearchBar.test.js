import React from "react";
import { render } from "@testing-library/react";
import SearchBar from "./SearchBar";

// smoke test
it("renders without crashing", function(){
    render(<SearchBar />);
});

// snapshot
it("matches snapshot", function() {
    const { asFragment } = render(<SearchBar />);
    expect (asFragment()).toMatchSnapshot();
});