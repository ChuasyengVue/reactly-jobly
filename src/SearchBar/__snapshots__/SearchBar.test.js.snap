// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <input
        name="search"
        placeholder="Enter search term.."
        type="text"
        value=""
      />
      <button
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
</DocumentFragment>
`;
